pkg/ui/workspaces/cluster-ui/src/api/contentionApi.ts

```diff
@@ -0,0 +1,173 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+import {
+  executeInternalSql,
+  LARGE_RESULT_SIZE,
+  LONG_TIMEOUT,
+  sqlApiErrorMessage,
+  SqlExecutionRequest,
+  sqlResultsAreEmpty,
+} from "./sqlApi";
+import { ContentionDetails } from "src/insights";
+import moment from "moment";
+
+export type ContentionFilters = {
+  waitingTxnID?: string;
+  waitingStmtID?: string;
+  start?: moment.Moment;
+  end?: moment.Moment;
+};
+
+export type ContentionResponseColumns = {
+  waiting_txn_id: string;
+  waiting_txn_fingerprint_id: string;
+  collection_ts: string;
+  contention_duration: string;
+  blocking_txn_id: string;
+  blocking_txn_fingerprint_id: string;
+  waiting_stmt_id: string;
+  waiting_stmt_fingerprint_id: string;
+  schema_name: string;
+  database_name: string;
+  table_name: string;
+  index_name: string;
+  key: string;
+};
+
+export async function GetContentionDetailsApi(
```

Comment:
nit: I don't believe we capitalize functions even if they're exported for TS. Only type names / classes / components.